### PR TITLE
Adding big integer field and eth address field (introduce text-input-filter)

### PR DESCRIPTION
**Pull request by tintinthong**
This PR adds big integer field and eth address field
- in addition adds the `text-input-filter`
- supporting test

https://github.com/cardstack/boxel/assets/8165111/4956ed68-8210-48a0-a7e8-a67e265019d9

PS: https://github.com/cardstack/boxel/pull/461 pr fixes the issue of cursor jumping to the end

### CODE REVIEW

File: packages/host/tests/integration/components/input-field-test.gts [hunk 1]
```diff
@@ -0,0 +1,308 @@
+import { module, test } from 'qunit';
```
Comment (tintinthong) [inline packages/host/tests/integration/components/input-field-test.gts]: Here I add a new set of test that checks input validation of BigInt. These test should be generalised when adding the resource

File: packages/host/tests/integration/components/serialization-test.gts [hunk 2]
```diff
@@ -4416,4 +4418,146 @@ module('Integration | serialization', function (hooks) {
       });
     });
   });
+
+  module('base cards', function () {
```
Comment (tintinthong) [inline packages/host/tests/integration/components/serialization-test.gts:4426]: I added a custom block within serialization test. This is particurly useful for base cards with custom serialize and deserialize

File: packages/base/big-integer.gts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    />
+  </template>
+
+  validatorEditor: TextInputFilter<bigint> = new TextInputFilter(
+    () => this.args.model,
+    (inputVal) => this.args.set(inputVal),
+    _deserialize,
+    _serialize
+  );
+}
```
Comment (tintinthong) [inline packages/base/big-integer.gts]: This is typical usage on how to use the text input filter

File: packages/base/ethereum-address.gts [hunk 4]
```diff
@@ -20,41 +27,65 @@ function isChecksumAddress(address: string): boolean {
...
+      @invalid={{this.textInputFilter.isInvalid}}
+    />
+  </template>
+
+  textInputFilter: TextInputFilter<string> = new TextInputFilter(
+    () => this.args.model,
+    (inputVal) => this.args.set(inputVal),
+    _deserialize
+  );
+}
```
Comment (tintinthong) [inline packages/base/ethereum-address.gts:73]: Here is another example of usage of textInputFilter

File: packages/base/big-integer.gts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    />
+  </template>
+
+  textInputFilter: TextInputFilter<bigint> = new TextInputFilter(
+    () => this.args.model,
+    (inputVal) => this.args.set(inputVal),
+    _deserialize,
+    _serialize
+  );
+}
```
Comment (tintinthong) [inline packages/base/big-integer.gts:62]: Here is example of instantiating textInputFilter

File: packages/base/text-input-filter.gts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+ * - If the text input cannot be deserialized, the state is marked as invalid, and an appropriate error message is displayed. This reacts as the user types.
+ * - If the model of the field is updated independently of the text input (e.g., through another process or operation),
+ *   the input text should be updated accordingly to reflect the changed value.
+ *
+ * Usage:
+ * Instances of this class are typically created within a component and often used in conjunction with the BoxelInput component.
+ *
+ * Examples:
+ * See the implementation in `ethereum-address.ts` and `big-integer.ts` for practical usage.
+ */
```
Comment (tintinthong) [inline packages/base/text-input-filter.gts:27]: This describes the text input filter class

File: packages/host/tests/integration/components/serialization-test.gts [hunk 7]
```diff
@@ -4416,4 +4418,146 @@ module('Integration | serialization', function (hooks) {
...
+        assert.strictEqual(
+          serialized?.data?.attributes?.someComputed,
+          (BigInt('1') + BigInt('2')).toString()
+        );
+      });
+    });
+
+    module('ethereum address card', function () {});
+    //..and more
+  });
```
Comment (tintinthong) [inline packages/host/tests/integration/components/serialization-test.gts]: will add this in another PR

**Review by ef4 — APPROVED** [review]

**Review by burieberry — APPROVED** [review]

**Pull request merged**